sklearn/_build_utils/__init__.py

```diff
@@ -33,3 +35,36 @@ def atlas_not_found(blas_info_):
         cblas_libs = blas_info.pop('libraries', [])
 
     return cblas_libs, blas_info
+
+
+def get_cython_source(filename):
+    is_c_filename = filename.endswith('.c')
+    is_cpp_filename = filename.endswith('.cpp')
+
+    # files in src are .c and .cpp files that are not cython-generated
+    if 'src/' in filename and (is_c_filename or is_cpp_filename):
```

Comment:
This is the most brittle part of the PR I think. I don't know whether there is a better way of doing it, suggestions welcome.